CMakeLists.txt

```diff
@@ -4,6 +4,8 @@ cmake_minimum_required(VERSION 3.5)
 # disable testsuite in that case
 if(NOT DEFINED PROJECT_NAME)
     set(NOT_SUBPROJECT ON)
+else()
+    set(NOT_SUBPROJECT OFF)
```

Comment:
I think this would be better off as

`set(NOT_SUBPROJECT NOT DEFINED PROJECT NAME)`, and later turning `if(NOT_SUBPROJECT)` into `if(${NOT_SUBPROJECT})`.

Reply:
Could you please say why?

I can think of a number of downsides of that:

* It means updating all the uses of the variable need to be updated, increasing the scope of the PR
* It requires anyone making future edits to this project to notice, if they were making edits that used this variable (e.g. if they were updating an old fork), that they would need to update a usage of a variable whose use had been changed in this PR
* The [current usage is idiomatic](https://github.com/search?q=%22NOT_SUBPROJECT%22+filename%3ACMakeLists.txt&type=Code) in lots of CMake projects - to use something different risks people making errors if ever making future edits in the project

Reply:
My thinking here is that there are two ways to go about this.

* The way that is currently in the PR. In any common programming language, it would be a terrible style, in CMake it is less so, because it doesn't really handle boolean expressions in a way one would expect.
* The suggested way, which is less commonly in use, but more closely follows standard programming style in different languages. (CMake also uses it, i.e. for `cmake_dependent_option`)

However, I am willing to accept either style, as there is actually a much more important question to ask: why is the change necessary at all? `if` should already interpret empty string as false, and dereferencing an empty variable provides an empty string. Is the problem here that the same-named variable is defined by the including project?